A31-xds-routeaction-support.md

```diff
@@ -150,41 +150,45 @@ cluster name provided to it, which should not be possible when following the
+Only some xDS timeout-related settings will be supported:
+
+- [`RouteActions.max_stream_duration.max_stream_duration`](https://github.com/envoyproxy/envoy/blob/6f2ad057f8655e2688a195337f7520dbd77015fa/api/envoy/config/route/v3/route_components.proto#L775)
+- [`RouteActions.max_stream_duration.grpc_max_timeout`](https://github.com/envoyproxy/envoy/blob/6f2ad057f8655e2688a195337f7520dbd77015fa/api/envoy/config/route/v3/route_components.proto#L781)
+- [`max_stream_duration`](https://github.com/envoyproxy/envoy/blob/6f2ad057f8655e2688a195337f7520dbd77015fa/api/envoy/config/core/v3/protocol.proto#L95) from the [`HttpConnectionManager.common_http_options`](https://github.com/envoyproxy/envoy/blob/6f2ad057f8655e2688a195337f7520dbd77015fa/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L289) as a default if `RouteActions.max_stream_duration.max_stream_duration` is unset.
+
+Notably, gRPC will not support
+[`RouteActions.timeout`](https://github.com/envoyproxy/envoy/blob/6f2ad057f8655e2688a195337f7520dbd77015fa/api/envoy/config/route/v3/route_components.proto#L952)
+or
+[`RouteActions.max_stream_duration.grpc_timeout_offset`](https://github.com/envoyproxy/envoy/blob/6f2ad057f8655e2688a195337f7520dbd77015fa/api/envoy/config/route/v3/route_components.proto#L789).
 The `timeout` field begins the timeout counter from a point in time
 incompatible with gRPC's timeout model.
-`max_stream_duration.grpc_timeout_offset` is used in Envoy for ensuring the
+`RouteActions.max_stream_duration.grpc_timeout_offset` is used in Envoy for ensuring the
 proxy detects timeouts before the gRPC client cancels the RPC due to its
 deadline, so supporting it directly in the client is unnecessary and would be
 counter-productive.
 
-If `max_stream_duration.grpc_max_timeout` is not set, the
-`max_stream_duration.max_stream_duration` field will specify the maximum RPC
-timeout for this route, with a default of unlimited.
-
-If `max_stream_duration.grpc_max_timeout` is present, then
-`max_stream_duration.max_stream_duration` is ignored and
-`max_stream_duration.grpc_max_timeout` limits the maximum timeout for RPCs on
-this route instead.  A value of 0 indicates the application's deadline is used
-without modification.
-
-In all cases, the RPC timeout set by the application may never be
-exceeded due to these settings.  For examples, see the following
-table:
+If `RouteActions.max_stream_duration.grpc_max_timeout` is not set, the
+`RouteActions.max_stream_duration.max_stream_duration` field will specify the
+maximum RPC timeout for this route (or fall-back to the HTTP Connection Manager
+setting), with a default of unlimited.
+
+If `RouteActions.max_stream_duration.grpc_max_timeout` is present, then
+`RouteActions.max_stream_duration.max_stream_duration` is ignored and
+`RouteActions.max_stream_duration.grpc_max_timeout` limits the maximum timeout
+for RPCs on this route instead.  A value of 0 indicates the application's
```

Comment:
I guess this means that setting *either* `max_stream_duration` to 0 or `grpc_max_timeout` to 0 will effectively not specify any default timeout.

Should we include rows in the table below for `max_stream_duration` set to 0?

Reply:
Missed this somehow.

I'm actually not sure how `max_stream_duration` = 0 would/should behave.  Theoretically it should timeout instantly since there is no need for "0" to represent "infinite" AFAICT.

Reply:
Follow-up: looks like "0" disables the timer.

Reply:
Ah, you're right -- I was thinking it was an int, so unset was the same as 0.  But the field is actually a `google.protobuf.Duration`, so I think you're right.

Reply:
Oh, looks like you changed it to say that 0 is unlimited.  Please check what Envoy does -- we should be consistent with it here.